secure/key-manager/src/libra_interface.rs

```diff
@@ -0,0 +1,217 @@
+                    account
+                ))
+            }),
+            Err(e) => Err(Error::UnknownError(format!("{:?}", e))),
+        }
+    }
+
+    /// Fetches the ConfigurationResource associated with a specific account.
+    fn get_configuration_resource(
+        &self,
+        account: AccountAddress,
+    ) -> Result<ConfigurationResource, Error> {
+        let configuration_resource = self
+            .get_account_state(account)?
+            .get_configuration_resource();
+        match configuration_resource {
+            Ok(configuration_resource) => configuration_resource.ok_or_else(|| {
+                Error::DataDoesNotExist(format!(
+                    "ConfigurationResource not found for account: {:?}",
+                    account
+                ))
+            }),
+            e => Err(Error::UnknownError(format!("{:?}", e))),
+        }
+    }
+
+    /// Fetches the ValidatorConfigResource associated with a specific account.
+    fn get_validator_config_resource(
+        &self,
+        account: AccountAddress,
+    ) -> Result<ValidatorConfigResource, Error> {
+        let config_resource = self
+            .get_account_state(account)?
+            .get_validator_config_resource();
+        match config_resource {
+            Ok(config_resource) => config_resource.ok_or_else(|| {
+                Error::DataDoesNotExist(format!(
+                    "ValidatorConfigResource not found for account: {:?}",
+                    account
+                ))
+            }),
+            e => Err(Error::UnknownError(format!("{:?}", e))),
+        }
+    }
+
+    /// Fetches the AccountResource associated with a specific account.
+    fn get_account_resource(&self, account: AccountAddress) -> Result<AccountResource, Error> {
+        let account_resource = self.get_account_state(account)?.get_account_resource();
+        match account_resource {
+            Ok(account_resource) => account_resource.ok_or_else(|| {
+                Error::DataDoesNotExist(format!(
+                    "AccountResource not found for account: {:?}",
+                    account
+                ))
+            }),
+            e => Err(Error::UnknownError(format!("{:?}", e))),
+        }
+    }
+
+    /// Fetches the LibraTimestampResource associated with a specific account.
+    fn get_libra_timestamp_resource(
+        &self,
+        account: AccountAddress,
+    ) -> Result<LibraTimestampResource, Error> {
+        let timestamp_resource = self
+            .get_account_state(account)?
+            .get_libra_timestamp_resource();
+        match timestamp_resource {
+            Ok(timestamp_resource) => timestamp_resource.ok_or_else(|| {
+                Error::DataDoesNotExist(format!(
+                    "LibraTimestampResource not found for account: {:?}",
+                    account
+                ))
+            }),
+            e => Err(Error::UnknownError(format!("{:?}", e))),
```

Comment:
maybe use a from instead? why would we get an error? what type of errors can occur here? maybe unexpected?

Reply:
Well, this ultimately comes from the "get_resource" call, which can return an error. I think this error would generally be unexpected (e.g., if there was LCS serialization problem), but it still needs to be handled..

Reply:
Maybe UnexpectedError then?

Reply:
Is "UnexpectedError" that different from "UnknownError"? 😄 I thought they were synonymous hence why I didn't want to add another error type for something that just means "we don't know what went wrong?"